### ISSUE TITLE
How to write a clob using utf-8 encoding charset

### ISSUE THREAD

**Issue by felipelds** [vrogier/ocilib#170]
I have  a linux machine with the environment variable NLS_LANG=.AL32UTF8 and I can't write an uft-8 string using the Clob object. 

        ocilib::Statement st(conn);
        st.Prepare("INSERT INTO clob_test(cod, message) VALUES(:v1, :v2)");

        int cod = 1;
        st.Bind(":v1", cod, ocilib::BindInfo::In);

        ocilib::Clob clob(conn);
        clob.Write("maçã");
        st.Bind(":v2", clob, ocilib::BindInfo::In);

        st.ExecutePrepared();

And I get the error:
`Error number 24811: ORA-24811: less data provided for writing than indicated`

I'm doing something wrong?

**Comment by vrogier** [comment]
Hi,

Sorry for a such later response.
I think there maybe an issue with Ocilib::Clob::Write() that might pass a wrong value for buffer size to OCI_LobWrite().
I will investigate this issue asap and come back to you.

Best regards,

Vincent

**Comment by felipelds** [comment]
I think the ocilib is assign the string size in bytes to char count and that
was causing the issue.

To avoid this wrong assignment I changed the OCI_LobWrite function as follows:

**if(OCI_BLOB == lob->type)**
{
    byte_count = len;
    ptr_count  = &byte_count;
}
else
{
    char_count = len;
    ptr_count  = &char_count;
}

to:

**if(OCI_BLOB == lob->type || OCILib.nls_utf8)**
{
    byte_count = len;
    ptr_count  = &byte_count;
}
else
{
    char_count = len;
    ptr_count  = &char_count;
}

It seems to work but I'm not sure if this is the best way to address the issue.

**Comment by vrogier** [comment]
Hi,

Sorry for late response. 
Yes, it might be a way to address the issue.
I will do further testing and will a commit a fix for it.
Will be part of 4.6. release.

Thanks for reporting issues :)

Best regards,

Vincent

**Comment by vrogier** [comment]
Wrongly closed. Will be closed once 4.6.0 will be merged to master

**Comment by arenzaria** [comment]
I am currently facing the same issue although it seems to be marked as solved.
Am I missing something? Maybe a special define needed?

oracle instantclient version: 21_6
NLS_LANG AMERICAN_AMERICA.AL32UTF8
CHARSET ANSI

It works fine when I remove the special character...

```
CREATE TABLE test_table (
     id NUMBER PRIMARY KEY,
     langText CLOB
);

ocilib::Statement st(m_connection);
st.Prepare("INSERT INTO test_table (id, langText) VALUES (1, :langText)");

ocilib::ostring langText = "Test LangtextÖ";
ocilib::Clob clob{m_connection};
clob.Write(langText);
st.Bind(":langText", clob, ocilib::BindInfo::BindDirectionValues::In);
  
st.ExecutePrepared();
```
